tests/observatory/platform/cli/test_generate_command.py

```diff
@@ -241,3 +251,576 @@ def test_write_rendered_template(self, mock_click_confirm):
+        self.assertEqual(local_dags.workflows_projects[0], DefaultWorkflowsProject.academic_observatory_workflows())
+        self.assertEqual(local_dags.workflows_projects[1], DefaultWorkflowsProject.oaebu_workflows())
+        print(local_dags.workflows_projects)
+
+        terraform_nodags = InteractiveConfigBuilder.build(backend_type=BackendType.terraform, workflows=[], oapi=False)
+        self.assertTrue(isinstance(terraform_nodags, TerraformConfig))
+        self.assertTrue(isinstance(terraform_nodags, ObservatoryConfig))
+        self.assertEqual(len(terraform_nodags.workflows_projects), 0)
+
+        terraform_dags = InteractiveConfigBuilder.build(
+            backend_type=BackendType.terraform,
+            workflows=["academic-observatory-workflows", "oaebu-workflows"],
+            oapi=False,
+        )
+        self.assertTrue(isinstance(terraform_dags, TerraformConfig))
+        self.assertEqual(len(terraform_dags.workflows_projects), 2)
+        self.assertEqual(terraform_dags.workflows_projects[0], DefaultWorkflowsProject.academic_observatory_workflows())
+        self.assertEqual(terraform_dags.workflows_projects[1], DefaultWorkflowsProject.oaebu_workflows())
+
+        self.assertTrue(m_backend.called)
+        self.assertTrue(m_observatory.called)
+        self.assertTrue(m_terraform.called)
+        self.assertTrue(m_google_cloud.called)
+        self.assertTrue(m_airflow_connections.called)
+        self.assertTrue(m_airflow_variables.called)
+        self.assertTrue(m_workflows_projects.called)
+        self.assertTrue(m_cloud_sql_database.called)
+        self.assertTrue(m_airflow_main_vm.called)
+        self.assertTrue(m_airflow_worker_m.called)
+        self.assertTrue(m_elasticsearch.called)
+        self.assertTrue(m_api.called)
+
+    @patch("observatory.platform.cli.generate_command.click.prompt")
+    def test_config_backend(self, m_prompt):
+        m_prompt.return_value = "staging"
+
+        config = ObservatoryConfig()
+        expected = Backend(type=BackendType.local, environment=Environment.staging)
+        InteractiveConfigBuilder.config_backend(config=config, backend_type=expected.type)
+        self.assertEqual(config.backend.type, expected.type)
+        self.assertEqual(config.backend.environment, expected.environment)
+
+        config = TerraformConfig()
+        expected = Backend(type=BackendType.terraform, environment=Environment.staging)
+        InteractiveConfigBuilder.config_backend(config=config, backend_type=expected.type)
+        self.assertEqual(config.backend.type, expected.type)
+        self.assertEqual(config.backend.environment, expected.environment)
+
+    @patch("observatory.platform.cli.generate_command.click.prompt")
+    def test_config_observatory_filled_keys(self, m_prompt):
+        observatory = Observatory(
+            airflow_fernet_key="IWt5jFGSw2MD1shTdwzLPTFO16G8iEAU3A6mGo_vJTY=",
+            airflow_secret_key=("a" * 16,),
```

Comment:
```suggestion
            airflow_secret_key=("a" * 16),
```
With the comma included it generates a tuple instead of a string.